azurelinuxagent/ga/update.py

```diff
@@ -727,6 +727,13 @@ def __init__(self, path=None, pkg=None, host=None):
             if isinstance(e, ResourceGoneError):
                 raise
 
+            # The agent was improperly blacklisting versions due to a timeout
+            # encountered while downloading a later version. Errors of type
+            # socket.error are IOError, so this should provide sufficient
+            # protection against a large class of I/O operation failures.
+            if isinstance(e, IOError):
```

Comment:
ResourceGoneError causes the calling continue to on with the goal state processing, which I believe is correct.  IOError causes the calling code to stop processing, and handle the exception. I also believe this is correct, but if I have any concerns with this code it is that handling.